### PR TITLE
ref(manifests): Clearer manifest types and documentation

### PR DESCRIPTION
**Pull request by thomastaylor312**
This moves the manifests to their own `mod` with additional documentation. I also fixed some issues with places where we needed a map instead of a concrete struct

### CODE REVIEW

File: src/model.rs [hunk 1]
```diff
@@ -2,129 +2,172 @@ use std::collections::BTreeMap;
...
 }
 
+/// Properties for linkdefs
 #[derive(Debug, Serialize, Deserialize)]
 pub struct LinkdefProperty {
+    /// The target this linkdef applies to. This should be the name of an actor component
     pub target: String,
-    pub values: LinkdefValueItems,
+    /// Values to use for this linkdef
+    pub values: BTreeMap<String, String>,
```
Comment (brooksmtownsend) [inline src/model.rs:136]: nit: we should skip serializing this if empty

File: src/model.rs [hunk 2]
```diff
@@ -2,129 +2,172 @@ use std::collections::BTreeMap;
...
-#[derive(Debug, Serialize, Deserialize)]
-pub struct ComponentItem {
+pub struct Spread {
+    /// The name of this spread requirement
     pub name: String,
-    #[serde(rename = "type")]
-    pub component_type: ComponentType,
-    pub properties: Properties,
+    /// An arbitrary map of labels to match on for scaling requirements
+    pub requirements: BTreeMap<String, String>,
```
Comment (brooksmtownsend) [inline src/model.rs:154]: nit: skip if empty

File: src/model.rs [hunk 3]
```diff
@@ -2,129 +2,172 @@ use std::collections::BTreeMap;
...
-}
-
-#[derive(Debug, Serialize, Deserialize)]
-pub struct Manifest {
-    #[serde(rename = "apiVersion")]
-    pub api_version: String,
-    pub kind: String,
-    pub metadata: Metadata,
-    pub spec: Components,
+    pub weight: Option<usize>,
```
Comment (brooksmtownsend) [inline src/model.rs:157]: nit: skip if none

**Review by brooksmtownsend — APPROVED** [review]
Looks great 😄 I left some nits around serde serializing, but I'm really not sure if they have an effect so I'm approving

File: src/model.rs (hunk 3 above)
Comment (thomastaylor312) [reply src/model.rs:157]: This is already there, the diff just makes it looks like it hasn't

**Pull request merged**